ClosedXML/Excel/Cells/XLCell.cs

```diff
@@ -434,12 +502,18 @@ public Object Evaluate(Boolean force = false)
 
         internal void SetInternalCellValueString(String cellValue)
         {
-            SetInternalCellValueString(cellValue, this.HasFormula);
+            SetInternalCellValueString(cellValue, validate: false, parseToCachedValue: this.HasFormula);
         }
 
-        private void SetInternalCellValueString(String cellValue, Boolean parseToCachedValue)
+        private void SetInternalCellValueString(String cellValue, Boolean validate, Boolean parseToCachedValue)
         {
+            if (validate)
+            {
+                if (cellValue.Length > 32767) throw new ArgumentException("Cells can hold a maximum of 32,767 characters.");
```

Comment:
BTW, `ArgumentOutOfRangeException` would be more specific and it is inherited from `ArgumentException` so it is safe to replace it (no external code would be broken). Same for dates.
Up to you.